src/main/java/lotto/enums/Rank.java

```diff
@@ -35,20 +36,29 @@ public static Rank getRank(int numberOfMatch) {
                 .orElse(LOSE);
     }
 
-    public Prize getPrize() {
-        return prize;
+    public static Rank getRank(int numberOfMatch, boolean hasBonusNumber) {
+        int rankKey = Objects.hash(numberOfMatch, hasBonusNumber);
+        return Optional.ofNullable(ranks.get(rankKey))
```

Comment:
```suggestion
        return ranks.getOrDefault(rankKey, LOSE);
```
`Map.getOrDefault(key, defaultValue)` 를 사용하시면 `Optional`로 포장하지 않아도 됩니다.

Reply:
오 이런 좋은 방법이 있었네요!